### PR TITLE
Show raster scale and offset in raster properties (fix #47673)

### LINKED ISSUE #47673: QGIS do not show if a raster have a scale factor
### What is the bug or the crash?

While investigating for https://github.com/qgis/QGIS/issues/47656 the following surfaced:

1) if loading a raster that has scaling (sample attached)

[Uploading RT_LC08_L1TP_218072_20211002_20211002_01_RT_2021-10-02_B2.zip…]()

gdalinfo:

```
...
Band 1 Block=765x5 Type=UInt16, ColorInterp=Gray
  Min=80.000 Max=2678.000
  Minimum=80.000, Maximum=2678.000, Mean=317.152, StdDev=126.433
  NoData Value=65535
  Offset: 0,   Scale:0.0001
...
```

then QGIS do not show anywhere about it. Not in the legend and not in the layer properties, leading to much confusion:

image

Moreover without any indication about scaling, more confusion arise if the user use this type of rasters for further processing.

QGIS and SAGA do take into account scaling, while GDAL and GRASS don't (for the last two the user is expected to unscale the raster if needed https://gdal.org/programs/gdal_translate.html#cmdoption-gdal_translate-unscale).

### Steps to reproduce the issue

See above.

### Versions

3.22.4
3.24.0

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by alexbruy**
## Description
While raster scale and offset are taken into account while reading raster data, these values are not reported in the "Information" tab of the layer properties dialog.

Fixes #47673.

### CODE REVIEW

**Comment by alexbruy** [comment]
@gioman hope this will make you a bit more happy.

**Comment by gioman** [comment]
> @gioman hope this will make you a bit more happy.

@alexbruy 🫂 🇺🇦

File: src/core/providers/gdal/qgsgdalprovider.cpp [hunk 1]
```diff
@@ -557,6 +557,9 @@ QString QgsGdalProvider::htmlMetadata()
       QStringList categories = QgsOgrUtils::cStringListToQStringList( GDALcategories );
       myMetadata += QgsHtmlUtils::buildBulletList( categories );
     }
+    QStringList scaleOffset;
+    scaleOffset << QStringLiteral( "Scale: %1, Offset: %2" ).arg( bandScale( i ) ).arg( bandOffset( i ) );
+    myMetadata += QgsHtmlUtils::buildBulletList( scaleOffset );
```
Comment (nyalldawson) [inline src/core/providers/gdal/qgsgdalprovider.cpp:562]: shouldn't this be two separate items in the list, so they each get a bullet point?
i.e. something like:

```suggestion
    myMetadata += QgsHtmlUtils::buildBulletList( QStringList{ {
           QObject::tr("Scale: %1").arg( bandScale( i ) ),
           QObject::tr("Offset: %1").arg( bandOffset( i ) ),
    }) );
```

Comment (alexbruy) [reply src/core/providers/gdal/qgsgdalprovider.cpp:562]: I have no strong opinion here. `gdalinfo` outputs this values as a single line and I decided to apply the same approach.

**Review by nyalldawson — APPROVED** [review]

**Comment by agiudiceandrea** [comment]
@alexbruy, the scale and offset metadata are separated from the others: is this expected?
image

I think it could be also useful to add the info about raw original band data type (which could be different from the raster data type displayed by QGIS if an offset != 0 or a scale != 1 are set).

Please see https://github.com/qgis/QGIS/issues/47400 for another metadata (NBITS) that could be useful to display.

**Pull request merged**